include/range/v3/at.hpp

```diff
@@ -40,7 +40,7 @@ namespace ranges
             auto operator()(Rng &&rng, range_difference_type_t<Rng> n) const ->
                 decltype(begin(rng)[n])
             {
-                if(n >= ranges::distance(rng))
+                if( (n < 0) or (n >= ranges::distance(rng)) )
```

Comment:
Note that, unlike its standard library analogue, this method accepts a `difference_type` (which is typically signed) rather than a a `size_type` (which is typically unsigned). Consequently, extended the verification to ensure the queried index is non-negative.